src/content/en/fundamentals/native-hardware/recording-video/index.md

```diff
@@ -0,0 +1,212 @@
+
+{% include "web/_shared/contributors/mscales.html" %}
+
+Many browsers now have the ability to access video and audio input from the
+user. However, depending on the browser it might be a full dynamic and inline
+experience, or it could be delegated to another app on the user's device.
+
+## Start simple and progressively
+
+The easiest thing to do is simply ask the user for a pre-recorded
+file. Do this by creating a simple file input element and adding
+an `accept` filter that indicates we can only accept video files and ideally we
+will get them directly from the camera.
+
+    <input type="file" accept="video/*" capture="camera">
+
+This method works on all platforms. On desktop it will prompt the user to
+upload a file from the file system (ignoring `capture="camera"`). In Safari
+on iOS it will open up the camera app, allowing you to record video and
+then send it back to the web page; on Android it will give the user the
+choice of which app to use record the video in before sending it back to the web
+page.
+
+Once the user has finished recording and they are back in the website, you
+need to somehow get ahold of the file data. You can get quick access by
+attaching an `onchange` event to the input element and then reading
+the `files` property of the event object.
+
+<pre class="prettyprint">
+&lt;input type="file" accept="video/*" capture="camera" id="recorder">
+&lt;video id="player" controls>&lt;/video>
+&lt;script>
+  var recorder = document.getElementById('recorder');
+  var player = document.getElementById('player')'
+
+  recorder.addEventListener('change', function(e) {
+    var file = e.target.files[0];
+    // Do something with the video file.
+    player.src = URL.createObjectURL(file);
+  });
+&lt;/script>
+</pre>
+
+Once you have access to the file you can do anything you want with it. For
+example, you can:
+
+* Attach it directly to a `<video>` element so that you can play it
+* Download it to the user's device
+* Upload it to a server by attaching to an `XMLHttpRequest`
+* Draw the frames into a canvas and apply filters to it
+
+Whilst using the input element method of getting access to video data is
+ubiquitous, it is the least appealing option. We really want to get access to
+the camera and provide a nice experience directly in the page.
+
+## Access the camera interactively
+
+Modern browsers can have a direct line to the camera allowing us to build
+experiences that are fully integrated with the web page and the user will never
+leave the browser.
+
+### Acquire access to the camera
+
+We can directly access the camera by using an API in the WebRTC
+specification called `getUserMedia()`. `getUserMedia()` will prompt the user for
+access to their connected microphones and cameras.
+
+If successful the API will return a `Stream` that will contain the data from
+either the camera or the microphone, and we can then either attach it to
+a `<video>` element or save it using the `MediaRecorder` API.
```

Comment:
I know I didn't include it in the previous article, we should just quickly mention that you can attach it to a WebRTC stream too.